### ISSUE TITLE
KeyError: 'name' processing name-server leaf-list

### ISSUE THREAD

**Issue by kecorbin** [cmoberg/nso-ansible-demo#1]
**UPDATE** I've tried this with NSO 4.5.1 and 4.5.3, as well as trying the netsim path outlined in README and am still stuck with similar issues. 

Great project, I'm trying to replicate this demo using real devices already registered in NSO and am having some issues.  This could be something simple that I'm missing, but I found the docs a bit confusing.

I'm using `ansible 2.5.0b1`
After capturing the output from the curl command and converting to yaml, there is a step. 

**Paste the output into verify-device-tmpl.yaml under the line with the device name. Remember to indent correctly.**

It's unclear to me where that is such that an error is not generated. I read this to mean that `verify-device-tmpl.yaml` should look like this

```
          device:
            - name: "{{ device }}"
              top:
                tailf-ncs:config:
                  tailf-ned-cisco-ios:EXEC:
                    operations:
                      default: /api/config/devices/device/headend/config/ios:EXEC/_operations/default
                      exec: /api/config/devices/device/headend/config/ios:EXEC/_operations/exec
                  tailf-ned-cisco-ios:cached-show:
                    version:
                      license:
                        level: ax
                        type: Default. No valid license found.
                      model: CSR1000V
                      version: 16.5.1b
```

However this produces error:

```
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "no child in /ncs:devices/device{headend} with name top. children ncs:name,ncs:address,ncs:port,ncs:remote-node,ncs:ssh,ncs:description,ncs:location,ncs:authgroup,ncs:device-type,ncs:device-profile,ncs:connect-timeout,ncs:read-timeout,ncs:write-timeout,ncs:ssh-keep-alive,ncs:trace,ncs:ned-settings,ncs:commit-queue,ncs:session-pool,ncs:no-overwrite,ncs:out-of-sync-commit-behaviour,choice-lsa,ncs:active-settings,ncs:live-status-protocol,ncs:state,ncs:source,ncs:capability,ncs:module,ncs:platform,ncs:config,ncs:live-status,ncs:rpc,ncs:netconf-notifications,ncs:service-list,ncs:snmp-notification-address,al:alarm-summary,ncs:copy-capabilities,ncs:find-capabilities,ncs:add-capability,ncs:apply-template,ncs:instantiate-from-other-device,ncs:compare-config,ncs:sync-from,ncs:sync-to,ncs:check-sync,ncs:check-yang-modules,ncs:connect,ncs:disconnect,ncs:ping,ncs:delete-config,ncs:scp-to,ncs:scp-from"}
```

Moving it to other indentation levels, generally produces key error on `top`
```
'dict object' has no attribute 'top'
```
TIA,
Kevin

**Comment by kecorbin** [comment]
If I remove 'top' from the generated yaml from json2yaml.py as well as removing the following section
```
name-server:
  - name: 10.10.10.1
  - name: 10.10.20.1

```
then all seems to work well, this sort of smells of https://github.com/ansible/ansible/issues/34985

**Comment by kecorbin** [comment]
Full traceback
```
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_module_nso_verify.py", line 190, in <module>
    main()
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_module_nso_verify.py", line 170, in main
    violations = nso_verify.main()
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_module_nso_verify.py", line 112, in main
    value_builder.build('', key, value)
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 326, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 350, in _build_dict
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 328, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 368, in _build_list
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 326, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 350, in _build_dict
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 326, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 350, in _build_dict
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 326, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 350, in _build_dict
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 326, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 350, in _build_dict
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 328, in build
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 354, in _build_list
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 379, in _build_key
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 461, in _get_type
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 456, in get_type
  File "/var/folders/c8/v4m5hlt12yg69740_f30ncch0000gn/T/ansible_w0r2rN/ansible_modlib.zip/ansible/module_utils/network/nso/nso.py", line 459, in get_type
KeyError: 'name'
```

**Comment by cmoberg** [comment]
Confirmed to be an issue in a combination of NSO and the module. Fix is underway and aiming to be part of 2.5. Will update here when things land.

**Comment by cmoberg** [comment]
Moving this issue to the new repository here in preparation for migrating:
 https://github.com/NSO-developer/nso-ansible-demo/issues/1
